File: haskell/pangram/src/Pangram.hs
module Pangram (isPangram) where

import Data.Char (toLower)
import Data.Set (fromList)

isPangram :: String -> Bool
isPangram text = length charSet == length ['a'..'z']
    where charSet = fromList (charsOnly text)

charsOnly :: [Char] -> [Char]
charsOnly [] = []
charsOnly (c:cs)
    | l `elem` ['a'..'z'] = l : charsOnly cs
    | otherwise           =     charsOnly cs
    where l = toLower c

File: go/pangram/pangram.go
package pangram

import (
	"strings"
)

const NumberOfLetters = len("abcdefghijklmnopqrstuvwxyz")

func IsPangram(input string) bool {
	var charMap = map[rune]bool{}

	for _, char := range strings.ToLower(input) {
		if char >= 'a' && char <= 'z' {
			charMap[char] = true
		}
	}

	return len(charMap) == NumberOfLetters
}
